src/ssmDocument/ssm/ssmClient.ts

```diff
@@ -0,0 +1,125 @@
+    const serverModule = extensionContext.asAbsolutePath(path.join('dist/src/ssmDocument/ssm/', 'ssmServer.js'))
+    // The debug options for the server
+    // --inspect=6010: runs the server in Node's Inspector mode so VS Code can attach to the server for debugging
+    const debugOptions = { execArgv: ['--nolazy', '--inspect=6010'] }
+
+    // If the extension is launch in debug mode the debug server options are use
+    // Otherwise the run options are used
+    const serverOptions: ServerOptions = {
+        run: { module: serverModule, transport: TransportKind.ipc },
+        debug: { module: serverModule, transport: TransportKind.ipc, options: debugOptions },
+    }
+
+    const documentSelector = [
+        { schema: 'file', language: 'ssm-json' },
+        { schema: 'untitled', language: 'ssm-json' },
+        { schema: 'file', language: 'ssm-yaml' },
+        { schema: 'untitled', language: 'ssm-yaml' },
+    ]
+
+    // Options to control the language client
+    const clientOptions: LanguageClientOptions = {
+        // Register the server for json documents
+        documentSelector,
+        initializationOptions: {
+            handledSchemaProtocols: ['file', 'untitled'], // language server only loads file-URI. Fetching schemas with other protocols ('http'...) are made on the client.
+            provideFormatter: false,
+        },
+        synchronize: {
+            fileEvents: workspace.createFileSystemWatcher('**/*.ssm.{json,yml,yaml}'),
+        },
+        middleware: {
+            workspace: {
+                didChangeConfiguration: () =>
+                    client.sendNotification(DidChangeConfigurationNotification.type, { settings: getSettings() }),
+            },
+        },
+    }
+
+    // Create the language client and start the client.
+    const client = new LanguageClient(
+        'ssm',
+        localize('ssm.server.name', 'SSM Document Language Server'),
+        serverOptions,
+        clientOptions
+    )
+    client.registerProposedFeatures()
+
+    const disposable = client.start()
+    toDispose.push(disposable)
+
+    const languageConfiguration: LanguageConfiguration = {
+        wordPattern: /("(?:[^\\\"]*(?:\\.)?)*"?)|[^\s{}\[\],:]+/,
+        indentationRules: {
+            increaseIndentPattern: /({+(?=([^"]*"[^"]*")*[^"}]*$))|(\[+(?=([^"]*"[^"]*")*[^"\]]*$))/,
+            decreaseIndentPattern: /^\s*[}\]],?\s*$/,
+        },
+    }
+    languages.setLanguageConfiguration('ssm-json', languageConfiguration)
+    languages.setLanguageConfiguration('ssm-yaml', languageConfiguration)
+
+    return client.onReady().then(() => {
+        client.onNotification(ResultLimitReachedNotification.type, message => {
+            window.showInformationMessage(
+                `${message}\nUse setting 'aws.ssmDocument.ssm.maxItemsComputed' to configure the limit.`
+            )
+        })
+    })
+}
+
+export async function deactivate(): Promise<any> {
+    return Promise.resolve(undefined)
+}
```

Comment:
What does this do?

Reply:
The example vscode lsp-server has this function. I will remove this since it seems that it is not called.